src/js/components/DataTable/Footer.js

```diff
@@ -25,34 +49,43 @@ const Footer = forwardRef(
       ...rest
     },
     ref,
-  ) => (
-    <StyledDataTableFooter ref={ref} fillProp={fill} pin={tablePin} {...rest}>
-      <TableRow>
-        {groups && (
-          <TableCell plain size="xxsmall" pad="none" verticalAlign="top" />
-        )}
-        {(selected || onSelect) && <TableCell background={background} />}
-        {columns.map(column => {
-          const pin = [];
-          if (tablePin) pin.push('bottom');
-          if (column.pin) pin.push('left');
-          return (
-            <Cell
-              key={column.property}
-              background={background}
-              border={border}
-              context="footer"
-              column={column}
-              datum={footerValues}
-              pad={pad}
-              pin={pin.length ? pin : undefined}
-              primaryProperty={primaryProperty}
+  ) => {
+    const theme = useContext(ThemeContext) || defaultProps.theme;
+    const pin = tablePin ? ['bottom'] : [];
```

Comment:
Should it be `: undefined` if not tablePin?

Reply:
calcBackground checks pin.length so it's expecting an array. The body of calcBackground was extracted from the original logic as-is.